### PR TITLE
[k8sclusterreceiver]: add node metadata as resource attributes

### LINKED ISSUE #27910: [receiver/k8scluster] Optionally add labels and annotations from node as resource attributes to node metrics
### Component(s)

receiver/k8scluster

### Is your feature request related to a problem? Please describe.

The k8scluster collects several node-related metrics and it would be helpful to be able to add node labels and annotations as resource attributes to these metrics.

### Describe the solution you'd like

The ability to enable the receiver to add node labels and annotations as resource attributes.  The labels/annotations should be taken as is, with no manipulation applied.

### Describe alternatives you've considered

_No response_

### Additional context

Closely related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22620, but the receiver should have the capability to add the attributes itself.

### PR DESCRIPTION
**Pull request by haoqixu**
**Description:** 
enable the receiver to add node labels and annotations as resource attributes

**Link to tracking Issue:**
resolve  #27910

### CODE REVIEW

File: receiver/k8sclusterreceiver/documentation.md [hunk 1]
```diff
@@ -440,6 +440,8 @@ Current status reason of the pod (1 - Evicted, 2 - NodeAffinity, 3 - NodeLost, 4
 | k8s.kubeproxy.version | The version of Kube Proxy running on the node. | Any Str | false |
 | k8s.namespace.name | The k8s namespace name. | Any Str | true |
 | k8s.namespace.uid | The k8s namespace uid. | Any Str | true |
+| k8s.node.annotations | The k8s node annotations. | Any Map | false |
+| k8s.node.labels | The k8s node labels. | Any Map | false |
```
Comment (jinja2) [inline receiver/k8sclusterreceiver/documentation.md:444]: So the attributes will look like
```
k8s.node.labels: Map({"beta.kubernetes.io/arch":"amd64","beta.kubernetes.io/os":"linux","foo":"bar-worker","kubernetes.io/arch":"amd64","kubernetes.io/hostname":"kind-worker","kubernetes.io/os":"linux"})
```

[From sem conv,](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute) attribute values can't be maps like this. Furthermore, I suspect most backends will just parse the map as string, making these attributes non filterable by the node's label/annotation key (they'll have to look for it in the value of the attr key `k8s.node.annotations`). And also diverges from how the labels/annotations are converted to string k/v attributes directly in k8sattr processor.

Comment (haoqixu) [reply receiver/k8sclusterreceiver/documentation.md]: Thanks for the comment. I update the code to flatten the node metadata into string kv attributes just like k8sattr processor.

Comment (dmitryax) [reply receiver/k8sclusterreceiver/documentation.md]: Given that we are moving it to the same representation as k8sattributes processor, I believe we should provide a configuration allowing users to filter/transform them into resource attributes similar way.

This is a good opportunity to start applying a standardized config interface for filtering. I brougth this issue some time ago https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161 and I think the SIG agreed on option B. 

@haoqixu do you want to apply this filtering maybe in this of next PR?

cc @TylerHelmuth

Comment (haoqixu) [reply receiver/k8sclusterreceiver/documentation.md]: Thanks for the information. I'd like to work on that. Where should we put the filter configuration? If we want to add a new config under `resource_attributes`, we need to enhance mdatagen to support that (like #25134).

Comment (haoqixu) [reply receiver/k8sclusterreceiver/documentation.md]: It seems that the filter code generated by `mdatagen` is used to filter the metrics by resource attributes and we cannot use it to filter/transform node labels and annotation into resource attributes. :thinking:  @dmitryax

**Pull request closed without merging**